src/PosthogAnalytics.ts

```diff
@@ -136,6 +136,11 @@ export class PosthogAnalytics {
     private authenticationType: Signup["authenticationType"] = "Other";
     private watchSettingRef?: string;
 
+    // Temporary flag until we can switch to the Rust SDK without restarting the app.
+    // Currently, you have to set up the flag then logout and login again to switch.
+    // On login the matrixClient is passed to the analytics object, and we can check the crypto backend version.
+    private hasLoggedInWithRustCrypto = false;
```

Comment:
**Note for reviewer**: Ideally we should just listen to the `feature_rust_crypto` settings, and this is what we will do when we will have migration in place. But until then you need to enable that feature then logout and login, that's why there is this flag.

It will be set when the current MatrixClient is passed to the analytics object (on login / page load / ...), the trick is to look at the crypto version.

I am not sure if it's right to listen to the setting now as it's not yet working. Maybe it should for now just check the crypto version from matrix client for now, and then later when we have migration we use the setting?
Let me know and I will update